### PR TITLE
AB#92111 AB#92112 Add homes number view with logic

### PR DESCRIPTION
**Pull request by dawid-pawulski**

### CODE REVIEW

File: src/HE.Investments.Common/Domain/ValueObjects/TheRequiredIntValueObject.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    protected TheRequiredIntValueObject(
+        string? value,
+        string fieldName,
+        string displayName,
+        int minValue = int.MinValue,
+        int maxValue = int.MaxValue)
+    {
+        if (value.IsNotProvided())
+        {
+            OperationResult.New()
```
Comment (Antek012) [inline src/HE.Investments.Common/Domain/ValueObjects/TheRequiredIntValueObject.cs:19]: You can use ThrowValidationError to simplify

File: src/HE.Investments.FrontDoor.Domain/Project/Repository/ProjectRepository.cs [hunk 2]
```diff
@@ -76,6 +77,7 @@ private ProjectEntity MapToEntity(FrontDoorProjectDto dto)
             supportActivityTypes: new SupportActivitiesMapper().Map(dto.ActivitiesinThisProject),
             affordableHomesAmount: ProjectAffordableHomesAmount.Create(new AffordableHomesAmountMapper().ToDomain(dto.AmountofAffordableHomes)),
             organisationHomesBuilt: dto.PreviousResidentialBuildingExperience.IsProvided() ? new OrganisationHomesBuilt((int)dto.PreviousResidentialBuildingExperience!) : null,
-            isSiteIdentified: dto.IdentifiedSite.IsProvided() ? new IsSiteIdentified(dto.IdentifiedSite) : null);
+            isSiteIdentified: dto.IdentifiedSite.IsProvided() ? new IsSiteIdentified(dto.IdentifiedSite) : null,
+            homesNumber: dto.NumberofHomesEnabled_Built.IsProvided() ? new HomesNumber((int)dto.NumberofHomesEnabled_Built!) : null);
```
Comment (Antek012) [inline src/HE.Investments.FrontDoor.Domain/Project/Repository/ProjectRepository.cs:81]: I know that this is not you code, but dto property name does not follow naming convention.

**Review by Antek012 — APPROVED** [review]

**Review by Carq — APPROVED** [review]

File: src/HE.Investments.Common/Domain/ValueObjects/TheRequiredIntValueObject.cs (hunk 1 above)
Comment (dawid-pawulski) [reply src/HE.Investments.Common/Domain/ValueObjects/TheRequiredIntValueObject.cs:19]: nice! done!

File: src/HE.Investments.FrontDoor.Domain/Project/Repository/ProjectRepository.cs (hunk 2 above)
Comment (dawid-pawulski) [reply src/HE.Investments.FrontDoor.Domain/Project/Repository/ProjectRepository.cs:81]: I passed the information on to Pawel

**Pull request merged**